src/components/TranslationProvider.tsx

```diff
@@ -0,0 +1,25 @@
+import React, { ReactElement, useEffect, useState } from "react";
+import { TranslationContext } from "../contexts/TranslationContext";
+import { LANGS } from "../types/TranslationTypes";
+import defaultLang from "../i18n/en.json";
+
+export default function TranslationProvider(props: { children: ReactElement }) {
+  const [config, changeConfig] = useState(defaultLang);
+  const [currentLang, changeLang] = useState(LANGS.en);
+
+  useEffect(() => {
+    async function loadLang() {
+      const data = (await import(`../i18n/${currentLang}.json`)) as {
+        default: typeof config;
+      };
+      changeConfig(data.default);
+    }
+    loadLang();
```

Comment:
Can you give me more details why have you chosen this way of language change detection? So you create async function that you call right after you create it. Should it really be async as you do not use its async nature when calling it?

Reply:
`await import("../i18n/${currentLang}.json")` is async stuff, so function should be async. Calling function right after creation is default code style for `useEffect`